Assets/MixedRealityToolkit/Providers/BaseBoundarySystem.cs

```diff
@@ -22,6 +22,36 @@ public abstract class BaseBoundarySystem : BaseCoreSystem, IMixedRealityBoundary
             ExperienceScale scale) : base(profile)
         {
             Scale = scale;
+            ReadProfile();
+        }
+
+        /// <summary>
+        /// Reads the visualization profile contents and stores the values in class properties.
+        /// </summary>
+        private void ReadProfile()
+        {
+            BoundaryHeight = BoundaryVisualizationProfile.BoundaryHeight;
```

Comment:
Something that I think we should be a bit more explicit about is how mutable profile state is and if changing a profile's individual properties at runtime should actually affect the running state of the system.

I *think* that I've seen a few places which have done either, but most of the new ones I've seen seem to err more on the side of this style.

Either way, I think it's fine to take a stance, but we should probably have some documentation somewhere (i.e. in our profiles documentation) about the intent that it's read at enable, and changing profile properties also requires that you either restart the system, or that you re-assign the profile to the same profile (to force it to reevaluate new state)

Reply:
Personally, i believe we should cache the profile contents at load time and allow the developer to go back to defaults by reloading the profile.

I agree with the need for documentation on this and will start a discussion thread to allow the community to provide feedback